src/webgpu/api/validation/render_pipeline/inter_stage.spec.ts

```diff
@@ -1,10 +1,337 @@
 export const description = `
-TODO: interface matching between vertex and fragment shader validation for createRenderPipeline:
-    - superset, subset, etc.
+Interface matching between vertex and fragment shader validation for createRenderPipeline:
+
+[1] TODO(dawn:1448) replace with t.device.limits.maxInterStageShaderVariables
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert, range } from '../../../../common/util/util.js';
 
 import { CreateRenderPipelineValidationTest } from './common.js';
 
-export const g = makeTestGroup(CreateRenderPipelineValidationTest);
+class InterStageMatchingValidationTest extends CreateRenderPipelineValidationTest {
+  getVertexStateWithOutputs(outputs: string[]): GPUVertexState {
+    return {
+      module: this.device.createShaderModule({
+        code: `
+        struct A {
+            ${outputs.join(',\n')},
+            @builtin(position) pos: vec4<f32>,
+        }
+        @vertex fn main() -> A {
+            var vertexOut: A;
+            vertexOut.pos = vec4<f32>(0.0, 0.0, 0.0, 1.0);
+            return vertexOut;
+        }
+        `,
+      }),
+      entryPoint: 'main',
+    };
+  }
+
+  getFragmentStateWithInputs(
+    inputs: string[],
+    hasBuiltinPosition: boolean = false
+  ): GPUFragmentState {
+    return {
+      targets: [{ format: 'rgba8unorm' }],
+      module: this.device.createShaderModule({
+        code: `
+        struct B {
+            ${inputs.join(',\n')},
+            ${hasBuiltinPosition ? '@builtin(position) pos: vec4<f32>' : ''}
+        }
+        @fragment fn main(fragmentIn: B) -> @location(0) vec4<f32> {
+            return vec4<f32>(1.0, 1.0, 1.0, 1.0);
+        }
+        `,
+      }),
+      entryPoint: 'main',
+    };
+  }
+
+  getDescriptorWithStates(
+    vertex: GPUVertexState,
+    fragment: GPUFragmentState
+  ): GPURenderPipelineDescriptor {
+    return {
+      layout: 'auto',
+      vertex,
+      fragment,
+    };
+  }
+}
+
+export const g = makeTestGroup(InterStageMatchingValidationTest);
+
+g.test('location,match')
+  .desc(
+    `Control case: creating render pipeline with input and output location match should succeed.`
```

Comment:
Ideally control cases can be parameterizations of each individual test to help verify the actual test function code is correct. Of course, in this case, the test function is pretty trivial since you've delegated everything to helpers, so this isn't that critical. But I think it can still work nicely here.

So in this case, `location,mismatch` might have two parameterizations:
```ts
[
  { v: ['@location(0) __: f32'], f: ['@location(0) __: f32'], _valid: true },
  { v: ['@location(0) __: f32'], f: ['@location(1) __: f32'], _valid: false },
  { v: ['@location(1) __: f32'], f: ['@location(0) __: f32'], _valid: false },
]
```
You could easily add more items too like:
`{ v: ['@location(0) __: f32', '@location(1) __: vec2<f32>'], f: ['@location(1) __: vec2<f32>', '@location(0) __: f32'], _valid: true }`
`{ v: ['@location(1) __: f32', '@location(0) __: vec2<f32>'], f: ['@location(0) __: vec2<f32>', '@location(1) __: f32'], _valid: true }`